client/src/tpu_client.rs

```diff
@@ -187,7 +187,9 @@ impl LeaderTpuCache {
     }
 
     fn fetch_slot_leaders(rpc_client: &RpcClient, start_slot: Slot) -> Result<Vec<Pubkey>> {
-        Ok(rpc_client.get_slot_leaders(start_slot, 2 * MAX_FANOUT_SLOTS)?)
+        let epoch_info = rpc_client.get_epoch_info()?;
+        let fanout = (2 * MAX_FANOUT_SLOTS).min(epoch_info.slots_in_epoch * 2);
```

Comment:
```suggestion
        let fanout = (2 * MAX_FANOUT_SLOTS).min(epoch_info.slots_in_epoch);
```
I don't think we want to multiple by 2 here right? You could be towards the end of an epoch and I think this fails if fetching slot leaders for current epoch + 2?

Reply:
yea. good point